ntoskrnl/io/iomgr/deviface.c

```diff
@@ -304,7 +305,189 @@ IoGetDeviceInterfaceAlias(IN PUNICODE_STRING SymbolicLinkName,
                           IN CONST GUID *AliasInterfaceClassGuid,
                           OUT PUNICODE_STRING AliasSymbolicLinkName)
 {
-    return STATUS_NOT_IMPLEMENTED;
+    UNICODE_STRING BaseKeyName = RTL_CONSTANT_STRING(L"\\Registry\\Machine\\System\\CurrentControlSet\\Control\\Class\\");
+    UNICODE_STRING AliasKeyName = RTL_CONSTANT_STRING(L"\\Registry\\Machine\\System\\CurrentControlSet\\Services\\");
+    UNICODE_STRING bipName, InLinkName;
+    UNICODE_STRING GuidString;
+    HANDLE InterfaceKey = NULL;
+    PKEY_FULL_INFORMATION fip = NULL;
+    PKEY_BASIC_INFORMATION bip = NULL;
+    PWCHAR pdest;
+    ULONG i, Size;
+    LONG ret;
+    NTSTATUS Status;
+
+    PAGED_CODE();
+
+    /* Convert alias GUID to a string */
+    Status = RtlStringFromGUID(AliasInterfaceClassGuid, &GuidString);
+    if (!NT_SUCCESS(Status))
+    {
+        DPRINT1("RtlStringFromGUID() failed with status 0x%08lx\n", Status);
+        goto cleanup;
+    }
+
+    BaseKeyName.Length = 0;
+    BaseKeyName.MaximumLength += GuidString.Length;
+    BaseKeyName.Buffer = ExAllocatePoolWithTag(NonPagedPool,
+                                               BaseKeyName.MaximumLength,
+                                               TAG_IO);
+    if (!BaseKeyName.Buffer)
+    {
+        DPRINT1("ExAllocatePoolWithTag() failed\n");
+        Status = STATUS_INSUFFICIENT_RESOURCES;
+        goto cleanup;
+    }
+
+    /* Concatenate alias GUID string with base key */
+    Status = RtlAppendUnicodeStringToString(&BaseKeyName, &GuidString);
```

Comment:
Not sure your concatenation works as you wish (What's the original value of BaseKeyName.Length? )

Reply:
It is actually initialized as `0`, but, as you said in my previous KS PR: https://github.com/reactos/reactos/pull/3390#discussion_r546111732, `RtlAppendUnicodeStringToString` seems to properly update this value. So I guess I don't need to fill in that manually.

Reply:
Since it's the first operation on the string you might as well use `RtlCopyUnicodeString` though, which is more straightforward.

Reply:
Fixed.